### PR TITLE
fixed athena handler 

### LINKED ISSUE #9861: [Bug]: Athena handler is unable to retrieve the tables from the database
### Short description of current behavior

Athena handler is unable to retrieve the tables from the database
for more details check the issue here
https://github.com/mindsdb/mindsdb/issues/9826

### Video or screenshots

{C42598EB-88BD-47C5-BE92-E7EF04DA1384}

### Expected behavior

_No response_

### How to reproduce the error

_No response_

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by minakshisharma197**
## Description

Fixes #9861 

## Type of change

(Please delete options that are not relevant)

- [x] 🐛 Bug fix (non-breaking change which fixes an issue)

## Verification Process

To ensure the changes are working as expected:

 - [ ]   Test Location: Specify the URL or path for testing.
 - [ ]   Verification Steps: Outline the steps or queries needed to validate the change. Include any data, configurations, or actions required to reproduce or see the new functionality.

## Additional Media:

- [x] I have attached a brief loom video or screenshots showcasing the new functionality or change.

{3927970C-64BE-4D7A-9CF1-5A2AEAF9AA32}

## Checklist:

- [x] My code follows the style guidelines(PEP 8) of MindsDB.
- [x] I have appropriately commented on my code, especially in complex areas.
- [x] Necessary documentation updates are either made or tracked in issues.
- [x] Relevant unit and integration tests are updated or added.

### CODE REVIEW

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by minakshisharma197** [comment]
I have read the CLA Document and I hereby sign the CLA

File: mindsdb/integrations/handlers/athena_handler/athena_handler.py [hunk 1]
```diff
@@ -181,23 +181,22 @@ def get_columns(self, table_name: str) -> StatusResponse:
...
-            str: Query execution status
-        """
         while True:
             response = self.connection.get_query_execution(QueryExecutionId=query_execution_id)
             status = response['QueryExecution']['Status']['State']
             if status in ['SUCCEEDED', 'FAILED', 'CANCELLED']:
                 return status
 
             check_interval = self.connection_data.get('check_interval', 0)
+            if isinstance(check_interval, str):
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/athena_handler/athena_handler.py:191]: Maybe check:
```
if isinstance(check_interval, str) and check_interval.strip().isdigit():
    check_interval = int(check_interval)
```

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]

File: mindsdb/integrations/handlers/athena_handler/athena_handler.py (hunk 1 above)
Comment (minakshisharma197) [reply mindsdb/integrations/handlers/athena_handler/athena_handler.py]: I checked it, and it’s working.

**Comment by ZoranPandovski** [comment]
recheckcla

**Comment by ZoranPandovski** [comment]
recheck

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
Please fix the flake8 warnings https://github.com/mindsdb/mindsdb/actions/runs/11384915301/job/31673658240?pr=9865#step:6:115

File: mindsdb/integrations/handlers/athena_handler/athena_handler.py [hunk 2]
```diff
@@ -181,23 +181,17 @@ def get_columns(self, table_name: str) -> StatusResponse:
...
             if status in ['SUCCEEDED', 'FAILED', 'CANCELLED']:
                 return status
 
             check_interval = self.connection_data.get('check_interval', 0)
+            if isinstance(check_interval, str) and check_interval.strip().isdigit():
+                check_interval = int(check_interval)
             if check_interval > 0:
                 time.sleep(check_interval)
-
     def _parse_query_result(self, result: dict) -> pd.DataFrame:
```
Comment (ritwickrajmakhal) [inline mindsdb/integrations/handlers/athena_handler/athena_handler.py:195]: You'll still get flake8 errors:
```bash
(mindsdb-venv) ritwickrajmakhal@ritwick-IdeaPad-3:~/mindsdb$ flake8 mindsdb/integrations/handlers/athena_handler
mindsdb/integrations/handlers/athena_handler/athena_handler.py:195:5: E301 expected 1 blank line, found 0
```

File: mindsdb/integrations/handlers/athena_handler/athena_handler.py [hunk 3]
```diff
@@ -181,23 +181,18 @@ def get_columns(self, table_name: str) -> StatusResponse:
         return self.native_query(query)
 
     def _wait_for_query_to_complete(self, query_execution_id: str) -> str:
-        """
```
Comment (ZoranPandovski) [inline mindsdb/integrations/handlers/athena_handler/athena_handler.py:184]: Why the comment was removed?

**Review by ZoranPandovski — CHANGES_REQUESTED** [review]
Please fix flake8 warning https://github.com/mindsdb/mindsdb/actions/runs/11558760403/job/32214290473?pr=9865#step:6:115.

File: mindsdb/integrations/handlers/athena_handler/athena_handler.py (hunk 3 above)
Comment (minakshisharma197) [reply mindsdb/integrations/handlers/athena_handler/athena_handler.py]: I accidentally removed the comment. Sorry!

**Comment by ZoranPandovski** [comment]
You need to solve the failing tests for flake8 warnings. Please check https://github.com/mindsdb/mindsdb/actions/runs/11577029045/job/32228876302?pr=9865#step:6:115

**Review by ZoranPandovski — APPROVED** [review]

**Pull request merged**